### PR TITLE
#185172119 implement change password

### PR DESCRIPTION
**Pull request by princoo**
#### What does this PR do?

-ensures a user changes the password to a new one
-ensure the new password is strong as the old one
- make sure the form is validated before making an API call

#### How should this be manually tested?
- login as a user
- on navigation go to change password
- enter the old ,new and confirm the password
- the new password must not be the same as the old password

#### What are the relevant pivotal tracker/Trello stories?

#185172119

### CODE REVIEW

File: src/pages/editAccount/main.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+const Changepassword = () => {
+  const [isPending, setisPending] = useState(false);
+  const oldPass = useRef<HTMLInputElement | null>(null);
+  const newPass = useRef<HTMLInputElement | null>(null);
+  const confirmPass = useRef<HTMLInputElement | null>(null);
+  const navigate = useNavigate();
+
+  async function handle() {
+    setisPending(true);
+    const divElement = document.getElementsByClassName('change-password-error')[0];
```
Comment (Kress2000) [inline src/pages/editAccount/main.tsx:18]: this is not a good approach to work within React TSX, you have a virtual document already. use Redux or useState to get the value and put them inside the TSX elements there down., what if a class is changed?

File: src/pages/editAccount/main.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      <form action="" method="post" name="signupForm" id={styles.signupForm}>
+        <h2 className={styles.formTitle}>Change Password</h2>
+        <p>Enter your new password below</p>
+        <div className={`${styles.error} change-password-error`} data-testid="error_div">
+          {' '}
+        </div>
+        <div className={styles.inputDiv}>
+          <Input
+            textId="in"
+            placeholder="OLD PASSWORD"
```
Comment (Kress2000) [inline src/pages/editAccount/main.tsx:55]: and here needs to be formik validations. user won't submit with fake or bad credentials

File: src/pages/home/home.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Link } from 'react-router-dom';
 import { useAppSelector, useAppDispatch } from '../../redux/hooks';
 import { increment } from '../../redux/slices/counterSlice';
 import { SetStateAction, useState } from 'react';
-import GoogleBtn from '../../components/googleLogin/button';
```
Comment (Kress2000) [inline src/pages/home/home.tsx:5]: This btn was used to logIn with google I think. why removed it?

**Review by Kress2000 — CHANGES_REQUESTED** [review]
check some comments there

File: src/components/button/button.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      data-testid={textId}
+      type={type}
+      disabled={loading_state}
+      onClick={onclick}
+      className={styles.button}
+    >
+      {loading_state ? <img src={loader} style={{ height: '25px' }} alt="loader" /> : value}
+    </button>
+  );
+};
```
Comment (munezerobagira) [inline src/components/button/button.tsx:25]: duplicate code from adrine PR

Comment (munezerobagira) [reply src/components/button/button.tsx:25]: https://github.com/atlp-rwanda/Team-Sostene-E-commerce-fe/blob/125efc0ac9a5101e84bd4b296ef60b407d0e90eb/src/components/reusables/Reusable.tsx

File: src/components/input/input.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <input
+      name={name}
+      data-testid={textId}
+      type={type}
+      placeholder={placeholder}
+      ref={innerRef}
+      className={`${styles.all}`}
+    />
+  );
+};
```
Comment (munezerobagira) [inline src/components/input/input.tsx:22]: check this one also

File: src/pages/home/home.tsx [hunk 6]
```diff
@@ -32,7 +32,7 @@ function Home() {
       <Link to={`/accounts/authenticate`} state={{ email }}>
         Click Here to go TFA
       </Link>
-      <GoogleBtn width="300" />
+      <Changepassword />
```
Comment (munezerobagira) [inline src/pages/home/home.tsx:35]: this can be moved from he   user profile drop down before logout

**Review by munezerobagira — CHANGES_REQUESTED** [review]

**Review by Kress2000 — APPROVED** [review]

**Review by Edcreation — APPROVED** [review]

**Review by munezerobagira — CHANGES_REQUESTED** [review]

File: src/pages/accounts/editAccount/changepass.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            type="password"
+            label="CONFIRM PASSWORD"
+            name="confirmPass"
+            id="confirmPassword"
+            textId="confirmpassword"
+            value={formik.values.confirmPass}
+            onChange={handleFieldChange}
+          />
+        </div>
+
```
Comment (munezerobagira) [inline src/pages/accounts/editAccount/changepass.tsx:108]: Display error below the text same here we can do it same here

**Review by munezerobagira — CHANGES_REQUESTED** [review]

**Review by Pacific911 — APPROVED** [review]

**Review by JosephNgabo — APPROVED** [review]

**Review by munezerobagira — APPROVED** [review]

**Pull request merged**